dms/models/directory.py

```diff
@@ -643,17 +637,23 @@ def write(self, vals):
             res = super().write(vals)
 
         if self and any(
-            field for field in vals if field in ["root_storage_id", "parent_id"]
+            field for field in vals if field in ["storage_id", "parent_id"]
```

Comment:
I still don't have this part clear, as it means you can change the storage in the parent directory, and be propagated to children, but no other action is taken, so the storage is inconsistent. Imagine you change from filestore to DB, but the files already contained are still in filestore with no transformation. Am I wrong?

Maybe we should just restrict the change of the storage.

Reply:
Correct, you should not be able to change the storage if it is of a different type.